stream/client/async_client.py

```diff
@@ -0,0 +1,276 @@
+        auth_token = self.create_jwt_token("activities", "*", feed_id="*")
+
+        data = {"changes": updates or []}
+
+        return await self.post("activity/", auth_token, data=data)
+
+    async def track_engagements(self, engagements):
+        auth_token = self.create_jwt_token("*", "*", feed_id="*")
+        await self.post(
+            "engagement/",
+            auth_token,
+            data={"content_list": engagements},
+            service_name="analytics",
+        )
+
+    async def track_impressions(self, impressions):
+        auth_token = self.create_jwt_token("*", "*", feed_id="*")
+        await self.post(
+            "impression/", auth_token, data=impressions, service_name="analytics"
+        )
+
+    async def og(self, target_url):
+        auth_token = self.create_jwt_token("*", "*", feed_id="*")
+        params = {"url": target_url}
+        return await self.get("og/", auth_token, params=params)
+
+    async def follow_stats(self, feed_id, followers_slugs=None, following_slugs=None):
+        auth_token = self.create_jwt_token("*", "*", feed_id="*")
+        params = {"followers": feed_id, "following": feed_id}
+
+        if followers_slugs:
+            params["followers_slugs"] = (
+                ",".join(followers_slugs)
+                if isinstance(followers_slugs, list)
+                else followers_slugs
+            )
+
+        if following_slugs:
+            params["following_slugs"] = (
+                ",".join(following_slugs)
+                if isinstance(following_slugs, list)
+                else following_slugs
+            )
+
+        return await self.get("stats/follow/", auth_token, params=params)
+
+    async def _make_request(
+        self,
+        method,
+        relative_url,
+        signature,
+        service_name="api",
+        params=None,
+        data=None,
+    ):
+        params = params or {}
+        data = data or {}
+        serialized = None
+        default_params = self.get_default_params()
+        params = self._check_params(params)
+        default_params.update(params)
+        headers = self.get_default_header()
+        headers["Authorization"] = signature
+        headers["stream-auth-type"] = "jwt"
+
+        if not relative_url.endswith("/"):
+            relative_url += "/"
+
+        url = self.get_full_url(service_name, relative_url)
+
+        if method.lower() in ["post", "put", "delete"]:
+            serialized = serializer.dumps(data)
+
+        async with self.session() as session:
+            async with session.request(
+                method,
+                url,
+                data=serialized,
+                headers=headers,
+                params=default_params,
+                timeout=self.timeout,
+                verify_ssl=False,
+            ) as response:
+
+                logger.debug(
+                    f"stream api call {response}, headers {headers} data {data}",
```

Comment:
We shouldn't leak the JWT into the logs. Can we either remove headers or filter out Authorization header?

Reply:
I removed `Authorization` from headers